src/Windows/R/Components/Impl/Resources.resx

```diff
@@ -446,37 +446,43 @@ If you want to delete all data on the remote machine you must delete your user p
     <value>Are you sure you want to remove the '{0}' connection?</value>
   </data>
   <data name="ContainerManager_ContainerServiceIsNotInstalled_FirstPart" xml:space="preserve">
-    <value>Docker service isn't installed. To start working with containers,</value>
+    <value>Docker service isn't installed. To start working with containers, </value>
   </data>
   <data name="ContainerManager_ContainerServiceIsNotInstalled_SecondPart" xml:space="preserve">
     <value>install Docker for Windows</value>
   </data>
   <data name="ContainerManager_ContainerServiceIsNotInstalled_ThirdPart" xml:space="preserve">
-    <value>and click</value>
+    <value> and click </value>
   </data>
   <data name="ContainerManager_ContainerServiceIsNotInstalled_FourthPart" xml:space="preserve">
     <value>refresh</value>
   </data>
+  <data name="ContainerManager_ContainerServiceIsNotInstalled_FifthPart" xml:space="preserve">
+    <value>.</value>
+  </data>
   <data name="ContainerManager_ContainerServiceIsNotRunning_FirstPart" xml:space="preserve">
     <value>Docker service isn't running.
-To start working with containers, run Docker for Windows and click</value>
+To start working with containers, run Docker for Windows and click </value>
```

Comment:
Some languages do not allow construction of sentences from parts. I.e. 'refresh' may or may not appear at the end of the sentence depending on the language.

Reply:
yes, that's why there is `ContainerManager_ContainerServiceIsNotRunning_FirstPart` that goes before refresh and `ContainerManager_ContainerServiceIsNotRunning_ThirdPart` that goes after refresh. In other languages any of those strings can be empty, which doesn't affect formatting in general.